native/membrane_videocompositor/src/elixir_bridge/elixir_structs.rs

```diff
@@ -308,7 +351,7 @@ impl TryInto<crate::scene::Scene> for Scene {
     type Error = SceneParsingError;
 
     fn try_into(self) -> Result<crate::scene::Scene, Self::Error> {
-        self.validate()?;
+        SceneValidator::new(&self).validate()?;
```

Comment:
What is the benefit of using a short-living struct instead of a single function operating on the scene directly?

Reply:
The `Scene::validate` function was defined in a `impl Scene` in a different module, which is quite unconventional and causes some unintuitive encapsulation behaviour. Since the struct only contains references to the `Scene`, it most likely compiles to the same code.

We could also just define a `validate(scene: &Scene)` function in the validation module, but I somehow like this solution better (the `SceneValidator` sort of 'prepares' the common parameters for the validation functions, which means they don't need to be passed in explicitly).

Reply:
`SceneValidator` indicates some OO actions and other versions o validations. 

I do not like this pattern, but it is not crucial. I will leave it up to you.